src/tactic/solve_by_elim.lean

```diff
@@ -55,7 +68,7 @@ As an optimisation, after we build the list of `tactic expr`s, we actually run t
 that do not in fact produce metavariables with a simple `return` tactic.
 -/
 meta def mk_assumption_set (no_dflt : bool) (hs : list simp_arg_type) (attr : list name) :
-  tactic (list (tactic expr)) :=
+  tactic (list (tactic expr) × tactic (list expr)) :=
```

Comment:
This type is definitely tricky to digest. I guess it's okay for now, considering the documentation. If it were any more complicated, I'd suggest bundling it into a structure `assumption_set` or something.